### PR TITLE
Cleanup

### PR DESCRIPTION
**Pull request by tobson**

### CODE REVIEW

**Comment by tobson** [comment]
@tberlok Let me know if you agree with these changes. I actually wanna make a few more changes before merging this into master, but this seems like a good time for a code review.

Also, could you perhaps have a look at the examples and make sure they're working?

File: skeletor/field.py [hunk 1]
```diff
@@ -62,70 +62,82 @@ def active(self, rhs):
...
+        # TODO: I suggest we get rid of this. The guard layes will be
+        # overwritten anyway by `copy_guards()`.
         self[:lby, :] = 0.0
         self[uby:, :] = 0.0
         self[:, ubx:] = 0.0
         self[:, :lbx] = 0.0
 
     def add_guards_vector(self):
+        "Add *vector* data from guard cells to corresponding active cells."
+        # TODO: We need to find a way of only having one `add_guards()` method
```
Comment (tberlok) [inline skeletor/field.py:118]: I actually have this on a branch on my laptop. Should be easy to add.

File: skeletor/field.py [hunk 2]
```diff
@@ -62,70 +62,82 @@ def active(self, rhs):
...
+        "Copy data to guard cells from corresponding active cells."
+
+        # Make sure boundaries aren't already set
+        # TODO: turn this into a warning. Other than communication overhead,
+        # calling this multiple times doesn't really cause any harm.
+        assert not self.boundaries_set, 'Boundaries are already set!'
 
-        msg = 'Boundaries are already set!'
-        assert(not self.boundaries_set), msg
         lbx = self.grid.lbx
```
Comment (tberlok) [inline skeletor/field.py:72]: I thought you wanted to get rid of this way writing it by setting `g = self.grid`
and then using `g.lbx` and so on.

File: skeletor/field.py [hunk 3]
```diff
@@ -147,15 +159,11 @@ def __new__(cls, grid, time=0.0, **kwds):
 
         obj = super().__new__(cls, grid, time=time, **kwds)
 
-        # Grid spacing
-        # TODO: this should be a property of the Grid class
-        dx = obj.grid.Lx/obj.grid.nx
```
Comment (tberlok) [inline skeletor/field.py:152]: I think I have already added `dx` and `dy` to the grid class?

File: skeletor/grid.py [hunk 4]
```diff
@@ -13,6 +12,7 @@ def __init__(self, nx, ny, comm, nlbx=0, nubx=2, nlby=0, nuby=1,
         self.ny = ny
 
         # Grid size in x- and y-direction
+        # TODO: Let's set Lx=Ly=1 by default
```
Comment (tberlok) [inline skeletor/grid.py:15]: Agreed.

File: skeletor/grid.py [hunk 5]
```diff
@@ -43,31 +45,17 @@ def __init__(self, nx, ny, comm, nlbx=0, nubx=2, nlby=0, nuby=1,
...
-        self.kstrt = comm.rank + 1
-        self.nvp = comm.size
-
-        # nypmx = size of particle partition, including guard cells, in y
-        # nxpmx = size of particle partition, including guard cells, in x
-        # nypmn = value of nyp
-        self.nypmx = self.nyp + self.nlby + self.nuby
-        self.nypmn = self.nyp
-        self.nxpmx = self.nx + self.nlbx + self.nubx
+        self.lbx = lbx
```
Comment (tberlok) [inline skeletor/grid.py:50]: This looks good now!

**Review by tberlok — APPROVED** [review]
I think this looks good.

File: skeletor/field.py (hunk 2 above)
Comment (tobson) [reply skeletor/field.py:72]: Yes, this is on my list of things I wanna change before merging into master.

File: skeletor/field.py (hunk 1 above)
Comment (tobson) [reply skeletor/field.py:118]: Yes, should be easy. I think the way of doing this that I suggest further below in the comment should work and is fairly general.

File: skeletor/field.py (hunk 3 above)
Comment (tobson) [reply skeletor/field.py:152]: Yes, those lines above are actually deleted now.

**Pull request merged**